server/lib/amplitude.js

```diff
@@ -0,0 +1,235 @@
+  } ],
+  [ /^([\w-]+).verification.success$/, {
+    group: GROUPS.email,
+    event: 'click'
+  } ]
+]);
+
+const NOP = () => {};
+
+const EVENT_PROPERTIES = {
+  [GROUPS.email]: mixProperties(mapEmailType, mapService),
+  [GROUPS.login]: mixProperties(mapEntrypoint, mapService),
+  [GROUPS.registration]: mixProperties(mapEntrypoint, mapService),
+  [GROUPS.settings]: mapEntrypoint,
+  [GROUPS.sms]: NOP
+};
+
+const USER_PROPERTIES = {
+  [GROUPS.email]: NOP,
+  [GROUPS.login]: mapUtmProperties,
+  [GROUPS.registration]: mapUtmProperties,
+  [GROUPS.settings]: mapNewsletterAction,
+  [GROUPS.sms]: NOP
+};
+
+module.exports = receiveEvent;
+
+function receiveEvent (event, data) {
+  if (! event || ! data) {
+    return;
+  }
+
+  const eventType = event.type;
+  let mapping = EVENTS[eventType];
+  let eventCategory;
+
+  if (! mapping) {
+    for (const [ key, value ] of FUZZY_EVENTS.entries()) {
+      const match = key.exec(eventType);
+      if (match) {
+        mapping = value;
+        if (match.length === 2) {
+          eventCategory = match[1];
+        }
+        break;
+      }
+    }
+  }
+
+  if (mapping) {
+    let group = mapping.group;
+    if (mapping.isDynamicGroup) {
+      group = group(eventCategory);
+      if (! group) {
+        return;
+      }
+    }
+
+    process.stderr.write(`${JSON.stringify({
+      op: 'amplitudeEvent',
+      time: event.time,
+      event_type: `${group} - ${mapping.event}`,
+      session_id: data.flowBeginTime,
+      event_properties: mapEventProperties(group, eventCategory, data),
+      user_properties: mapUserProperties(group, eventCategory, data),
+      app_version: APP_VERSION,
+      language: data.lang
+    })}\n`);
+  }
+}
+
+function mapEventProperties (group, eventCategory, data) {
+  return Object.assign({
+    device_id: marshallOptionalValue(data.deviceId)
+  }, EVENT_PROPERTIES[group](eventCategory, data));
+}
+
+function mapUserProperties (group, eventCategory, data) {
+  return Object.assign({
+    flow_id: marshallOptionalValue(data.flowId),
+  }, USER_PROPERTIES[group](eventCategory, data));
+}
+
+function marshallOptionalValue (value) {
+  if (value && value !== 'none') {
+    return value;
+  }
+}
+
+function mixProperties (...mappers) {
+  return (eventCategory, data) => Object.assign({}, ...mappers.map(m => m(eventCategory, data)));
+}
+
+function mapEmailType (eventCategory) {
+  const email_type = EMAIL_TYPES[eventCategory];
+  if (email_type) {
+    return { email_type };
+  }
+}
+
+function mapEntrypoint (eventCategory, data) {
+  const entrypoint = marshallOptionalValue(data.entrypoint);
+  if (entrypoint) {
+    return { entrypoint };
+  }
+}
+
+function mapService (eventCategory, data) {
+  const service = marshallOptionalValue(data.service);
+  if (service) {
+    return { service };
+  }
+}
+
+function mapNewsletterAction (eventCategory) {
+  const newsletter = NEWSLETTER_ACTIONS[eventCategory];
+  if (newsletter) {
+    return { newsletter };
```

Comment:
There is an outstanding question for @davismtl about what this property should be called. The taxonomy says `value` but I thought that might be too generic for a user property name. I've gone with `newsletter` for now but will update it as per feedback.

Reply:
I left a comment in the gdoc. sorry for delay.